internal/provider/serverless_cluster_data_source_test.go

```diff
@@ -0,0 +1,92 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	mockClient "github.com/tidbcloud/terraform-provider-tidbcloud/mock"
+	"github.com/tidbcloud/terraform-provider-tidbcloud/tidbcloud"
+	clusterV1beta1 "github.com/tidbcloud/tidbcloud-cli/pkg/tidbcloud/v1beta1/serverless/cluster"
+)
+
+func TestAccServerlessClusterDataSource(t *testing.T) {
+	serverlessClusterDataSourceName := "data.tidbcloud_serverless_cluster.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testServerlessClusterDataSourceConfig,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(serverlessClusterDataSourceName, "display_name", "test-tf"),
+					resource.TestCheckResourceAttr(serverlessClusterDataSourceName, "region.name", "regions/aws-us-east-1"),
+					resource.TestCheckResourceAttr(serverlessClusterDataSourceName, "endpoints.public.port", "4000"),
```

Comment:
why here is `endpoints.public.port` and below is `endpoints.public_endpoint.port`